### PR TITLE
Add [Binding] attribute to all core attributes.

### PR DESCRIPTION
**Pull request by ahmelsayed**
Probably should have a test for this, but not sure how to get all attributes that the host supports in a test.

### CODE REVIEW

**Comment by mamaso** [comment]
You could probably make a test similar to the public surface tests - find all public types, if attribute test against list of types excluded from 'binding'

File: src/Microsoft.Azure.WebJobs/DisableAttribute.cs [hunk 1]
```diff
@@ -23,6 +24,7 @@ namespace Microsoft.Azure.WebJobs
     /// </remarks>
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method | AttributeTargets.Parameter)]
+    [Binding]
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs/DisableAttribute.cs]: This one isn't a binding - remove

File: src/Microsoft.Azure.WebJobs/DisableAttribute.cs [hunk 2]
```diff
@@ -23,6 +24,7 @@ namespace Microsoft.Azure.WebJobs
...
     public sealed class DisableAttribute : Attribute
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/DisableAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs/NoAutomaticTriggerAttribute.cs [hunk 3]
```diff
@@ -24,6 +25,7 @@ namespace Microsoft.Azure.WebJobs
     /// RunAndBlock). Instead, they must be invoked manually using the Call method.
     /// </remarks>
     [AttributeUsage(AttributeTargets.Method)]
+    [Binding]
     public sealed class NoAutomaticTriggerAttribute : Attribute
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/NoAutomaticTriggerAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs/SingletonAttribute.cs [hunk 4]
```diff
@@ -18,6 +19,7 @@ namespace Microsoft.Azure.WebJobs
     /// </remarks>
     /// </summary>
     [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class, AllowMultiple = true)]
+    [Binding]
     public sealed class SingletonAttribute : Attribute
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/SingletonAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs [hunk 5]
```diff
@@ -13,6 +14,7 @@ namespace Microsoft.Azure.WebJobs
     /// is in that order.
     /// </remarks>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method | AttributeTargets.Parameter)]
+    [Binding]
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs/TimeoutAttribute.cs [hunk 6]
```diff
@@ -17,6 +18,7 @@ namespace Microsoft.Azure.WebJobs
     /// it to any async operations it initiates.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method)]
+    [Binding]
     public sealed class TimeoutAttribute : Attribute
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/TimeoutAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs/TraceLevelAttribute.cs [hunk 7]
```diff
@@ -19,6 +20,7 @@ namespace Microsoft.Azure.WebJobs
     /// invocations that fail.
     /// </remarks>
     [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    [Binding]
```
Comment (MikeStall) [inline src/Microsoft.Azure.WebJobs/TraceLevelAttribute.cs]: no

File: src/Microsoft.Azure.WebJobs.ServiceBus/ServiceBusAccountAttribute.cs [hunk 8]
```diff
@@ -13,6 +14,7 @@ namespace Microsoft.Azure.WebJobs
     /// is in that order.
     /// </remarks>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method | AttributeTargets.Parameter)]
+    [Binding]
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.ServiceBus/ServiceBusAccountAttribute.cs]: not a binding

**Pull request merged**